plugins/kotlin-as-java/src/main/kotlin/converters/KotlinCompanion.kt

```diff
@@ -0,0 +1,74 @@
+package org.jetbrains.dokka.kotlinAsJava.converters
+
+import org.jetbrains.dokka.links.Callable
+import org.jetbrains.dokka.model.*
+import org.jetbrains.dokka.model.properties.PropertyContainer
+
+private const val DEFAULT_COMPANION_NAME = "Companion"
+
+internal fun DObject?.staticFunctionsForJava(): List<DFunction> {
+    if (this == null) return emptyList()
+    return functions.filter { it.isJvmStatic }
+}
+
+/**
+ * @return properties that will be visible as static for java.
+ * See [Static fields](https://kotlinlang.org/docs/java-to-kotlin-interop.html#static-fields)
+ */
+internal fun DObject?.staticPropertiesForJava(): List<DProperty> {
+    if (this == null) return emptyList()
+    return properties.filter { it.isJvmField || it.isConst || it.isLateInit }
+}
+
+internal fun DObject?.companionInstancePropertyForJava(): List<DProperty> {
```

Comment:
Is there a reason this has to return a list? The name suggest that it's a single property as well

If you need to sum lists somewhere and don't wanna deal with nulls, you could wrap it in `listOfNotNull`

Reply:
True, I wouldn't say I liked it but was unable to find a better decision. I changed the return type to `DProperty ` and wrapped it in the suggested function on the use side.